src/widgets/tabs.rs

```diff
@@ -82,6 +91,12 @@ impl<'a> Tabs<'a> {
     pub fn new<T>(titles: Vec<T>) -> Tabs<'a>
     where
         T: Into<Line<'a>>,
+        // The `Tabs::new` function could be made more flexible by accepting `Iter: IntoIterator`,
+        // with `Iter::Item: Into<Line<'a>>`.  This may be desirable at some point, but would be a
+        // breaking change, as type inference could no longer use `Tabs::new` to infer the type
+        // into which an iterator is collected.  For example, `Tabs::new((0..4).map(|i|
+        // format("Tab{i}")).collect())` would not compile, because the return type of `.collect()`
+        // could no longer be inferred to be `Vec<_>`.
```

Comment:
> Sounds good. That was something I was wondering on, but figured I'd err on the side of backwards compatibility when it isn't a project I've contributed to before.

I've been fairly OK with breaking compatibility for things that are easy to fix like type inference. This is something that generally breaks a single line of code in the user's application, and there's no good way to deprecate it first. Other places we've tried to at least give a couple of releases notification with deprecated attributes (especially for changes that affect a lot of code).

Also, I'm rarely worried about performance of copying for this library even where it isn't optimized for release builds, mainly because for normal screen sizes even bad algorithms should generally have a fairly low N (~50 rows, ~200 cols), so for the most part optimization is noise (unless proven otherwise by benchmarks).

Reply:
Sounds good.  I've updated the PR to accept any `IntoIterator<Item: Into<Line>>`  for calls to `Tabs::new`, and updated the two examples that relied on the previous behavior for type inference.